### DISCUSSION TITLE
Espressif ESP Zigbee & IDF/Arduino Naming Conventions "Zigbee Model"

### DISCUSSION THREAD

**Discussion by elgerg** [Koenkk/zigbee2mqtt#23788] [Devices]
Hi all especially @Koenkk,

As the ESP-IDF and Arduino capabilities are now getting some really good attention and I specifically mean the Arduino wrapper work going on here:

https://github.com/P-R-O-C-H-Y/arduino-esp32/tree/feature/zigbee-library/libraries/Zigbee/examples

I expect that there will be a whole bunch of devices that people would like built in support for.

I think it would be worthwhile trying to come up with come sort of convention for 

Zigbee Manufacturer
Zigbee Model

If we say something along the lines of:

Zigbee Manufacturer - ESP-XX-Hx or ESP-XX-Cx
Where XX can be swapped out for the users preferred thing, IE ESP-ElgerG-H2

Then for model, we would need some sort of pattern, not really sure about that one, they currently seem to support (or will) Lights (On/Off and Dimmer) and Thermostat/Temp sensor. So maybe something like:

Zigbee Model - L-123-D-123-TH-TS

Then once split on - it would indicate that the device has:

"L" Light relays 1 2 and 3
"D" Dimmers/PWM 1 2 and 3 (or if it only had one then D-1)
"TH" Thermostat capabilities
"TS" Temp Sensor

In future maybe add in RGB dimmer.

Then expanding on that. Do any of you think this would mean we could include things without having to maintain multiple definitions each for the devices we create?

Really looking for some comments/thoughts.

Regards

**Comment by Koenkk** [comment]
I don't see the reason for any naming conventions, based on the clusters, z2m can automatically generate a definition for the device (https://github.com/Koenkk/zigbee-herdsman-converters/blob/master/src/lib/generateDefinition.ts). E.g. currently any bulb (that follows the standards) that you pair with z2m will work out-of-the-box.
